Content.Server/Chemistry/EntitySystems/SolutionInjectOnCollideSystem.cs

```diff
@@ -30,15 +46,33 @@ private void HandleInit(EntityUid uid, SolutionInjectOnCollideComponent componen
 
         private void HandleInjection(EntityUid uid, SolutionInjectOnCollideComponent component, StartCollideEvent args)
         {
+
             if (!EntityManager.TryGetComponent<BloodstreamComponent?>(args.OtherFixture.Body.Owner, out var bloodstream) ||
                 !_solutionsSystem.TryGetInjectableSolution(component.Owner, out var solution)) return;
 
+            if(component.CanPenetrateHelmet == false
+               && IsFaceBlocked(args.OtherFixture.Body.Owner , args.OurFixture.Body.Owner)
+               && component.CanPenetrateArmor == false) //TODO : Implement the armor check in another PR
+                return;
```

Comment:
```suggestion
            //TODO : Implement the armor check in another PR
            if (!component.CanPenetrateHelmet &&
               !component.CanPenetrateArmor &&
               IsFaceBlocked(args.OtherFixture.Body.Owner , args.OurFixture.Body.Owner))
                return;
```

These checks are done in order so you should always do the expensive one last.